packages/components/src/panel/header.tsx

```diff
@@ -0,0 +1,20 @@
+/**
+ * Internal dependencies
+ */
+import type { PanelHeaderProps } from './types';
+
+/**
+ * `PanelHeader` renders the header for the `Panel`.
+ * This is used by the `Panel` component under the hood,
+ * so it does not typically need to be used.
+ */
+function PanelHeader( { label, children }: PanelHeaderProps ) {
```

Comment:
Not going to add `PanelHeader` the Storybook for now because it shouldn't really be used independently. No usages in the repo either. If it weren't for back compat I would remove it from the public exports.

Reply:
Should we consider deprecating it to discourage further usage?

Reply:
The thing is, I doubt anyone is using it because there is no reason to 😆 So it's not that I want to discourage people from using it per se, it's more that I don't want to clutter our docs with something that is irrelevant to 99.9% of people.